pilot/cmd/pilot-agent/status/util/stats.go

```diff
@@ -38,7 +36,7 @@ const (
 )
 
 var (
-	readinessTimeout = env.RegisterDurationVar("ENVOY_READINESS_CHECK_TIMEOUT", time.Second*5, "").Get()
+	readinessTimeout = time.Second * 3 // Default Readiness timeout. It is set the same in helm charts.
```

Comment:
Curious about the original timeout reason, what is the normal latency?

Reply:
When Envoy is processing huge set of config from Pilot, people observed more than 5s and even up to 10s. That is why it was made configurable. Now we are caching the readiness state so this will only be executed till it is ready once - so mostly during initial startup. So conservatively I have set it to 3s

Reply:
sg